ranger-core/src/main/java/io/appform/ranger/core/finderhub/DynamicDataSource.java

```diff
@@ -0,0 +1,38 @@
+package io.appform.ranger.core.finderhub;
+
+import io.appform.ranger.core.model.Service;
+import lombok.NoArgsConstructor;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@NoArgsConstructor
+public class DynamicDataSource implements ServiceDataSource {
+    private final Set<Service> serviceCollection = ConcurrentHashMap.newKeySet();
```

Comment:
Check if service is comparable

Reply:
Service uses @Data annotation this include equalsandHashcode implementation